dojo/tools/cargo_audit/parser.py

```diff
@@ -24,23 +24,35 @@ def get_findings(self, filename, test):
             for item in data.get('vulnerabilities').get('list'):
                 advisory = item.get('advisory')
                 vuln_id = advisory.get('id')
-                title = advisory.get('title')
-                description = "\n".join([
-                    f"**Description:** `{advisory.get('description')}`",
-                    f"\n**Read more:** `{advisory.get('url')}`",
-                ])
+                if "categories" in advisory:
+                    categories = f"**Categories:** {', '.join(advisory['categories'])}"
+                else:
+                    categories = ''
+                description = categories + f"\n**Description:** `{advisory.get('description')}`"
+                if "functions" in item["affected"]:
```

Comment:
The second advisory in the test data with many vulnerabilities has `"affected": null,`. This might cause the problem in the unit test. You should check if `item["affected"]` is not None, before you use it.

Reply:
Added checking